eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -375,7 +376,8 @@ class
   ) =>
   ShelleyEraImp era
   where
-  initImpNES :: NewEpochState era
+  initImpNES ::
+    (Era era, MonadState (ImpTestState era) m) => m (NewEpochState era)
```

Comment:
I was thinking more about it and this turns out to be a lot more general function. So, let's rename it and change its semantics to reflect that:
```suggestion
  initImpTestState :: (MonadState (ImpTestState era) m, MonadGen m) => m ()
```

Important notes:

* `Era` is redundant
* `MonadGen` is useful, as mentioned in my other comment
* This action should not return anything, but instead should modify the state with all the initialization that needs to be done.